name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/AlphavantageQuoteFeed.java

```diff
@@ -98,59 +99,51 @@ public boolean updateLatestQuotes(Security security, List<Exception> errors)
         if (!rateLimiter.tryAcquire())
             throw new RateLimitExceededException(Messages.MsgAlphaVantageRateLimitExceeded);
 
-        String wknUrl = MessageFormat.format("https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY" //$NON-NLS-1$
+        String wknUrl = MessageFormat.format("/query?function=TIME_SERIES_INTRADAY" //$NON-NLS-1$
                         + "&symbol={0}&interval=1min&apikey={1}&datatype=csv&outputsize=compact", //$NON-NLS-1$
                         security.getTickerSymbol(), apiKey);
 
-        try
-        {
-            URL obj = new URL(wknUrl);
-            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
-            con.setConnectTimeout(1000);
-            con.setReadTimeout(20000);
+        WebAccess webAccess = WebAccess.builder().document("https", "www.alphavantage.co", wknUrl)//$NON-NLS-1$ //$NON-NLS-2$
+                        .build();
 
-            int responseCode = con.getResponseCode();
-            if (responseCode != HttpURLConnection.HTTP_OK)
-                throw new IOException(wknUrl + " --> " + responseCode); //$NON-NLS-1$
+        InputStream con = new ByteArrayInputStream(webAccess.getDocument().getBytes());
```

Comment:
Hier könntest Du den Code noch vereinfachen. Erst lädst Du den Text, dann wird er wieder in bytes umgewandelt, in eine InputStream gepackt, mit dem Scanner wieder ausgelesen und wieder in Text umgewandelt. All das ist ja jetzt zum Glück nicht mehr notwendig 😄

Reply:
Ich war erst am verzweifeln, da ich nicht verstehen wollte was du mir damit sagen wolltest. Erst bei näher Beschäftigung mit der Scanner Funktion habe ich den Hinweis verstanden. Da war was mit dem Baum und Wälder :man_facepalming:

Solved